druid-lookups/src/main/java/com/yahoo/maha/maha_druid_lookups/server/lookup/namespace/MahaLookupExprMacro.java

```diff
@@ -0,0 +1,126 @@
+package com.yahoo.maha.maha_druid_lookups.server.lookup.namespace;
+
+import com.google.common.base.Splitter;
+import com.google.inject.Inject;
+import com.yahoo.maha.maha_druid_lookups.query.lookup.MahaRegisteredLookupExtractionFn;
+import org.apache.druid.common.config.NullHandling;
+import org.apache.druid.java.util.common.IAE;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.math.expr.Expr;
+import org.apache.druid.math.expr.ExprEval;
+import org.apache.druid.math.expr.ExprMacroTable;
+import org.apache.druid.math.expr.ExprType;
+import org.apache.druid.query.lookup.LookupExtractorFactoryContainerProvider;
+import org.apache.druid.query.lookup.LookupReferencesManager;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MahaLookupExprMacro implements ExprMacroTable.ExprMacro
+{
+    private static final Logger LOG = new Logger(MahaLookupExprMacro.class);
+    private static final String FN_NAME = "maha_lookup";
+    private final LookupExtractorFactoryContainerProvider lookupExtractorFactoryContainerProvider;
+    private static final String SEPARATOR = ",";
+    private static final String KV_DEFAULT = "->";
+
+    @Inject
+    public MahaLookupExprMacro(final LookupExtractorFactoryContainerProvider lookupExtractorFactoryContainerProvider)
+    {
+        this.lookupExtractorFactoryContainerProvider = lookupExtractorFactoryContainerProvider;
+    }
+
+    @Override
+    public String name()
+    {
+        return FN_NAME;
+    }
+
+    @Override
+    public Expr apply(final List<Expr> args)
+    {
+        if (args.size() < 4) {
+            throw new IAE("Function[%s] must have at least 4 arguments", name());
+        }
+
+        final Expr arg = args.get(0);
+        final Expr lookupExpr = args.get(1);
+        final Expr columnExpr = args.get(2);
+        String columnStr = (String) columnExpr.getLiteralValue();
+        final Expr missingValueExpr = args.get(3);
```

Comment:
missing value onwards params are optional, you can handle the null checks that we are doing in SQL OP Conversion